OneCommunityGlobal/HGNRest
# Shengwei Enhancement: User Profile Page Performance

# Description
<img width="637" alt="pr2110_desc" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/69882989/8f512449-0352-432b-bf64-da9049eeecf2">


## Related PRS (if any):
To test this backend PR you need to checkout the [2110 ](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/2110) frontend PR.
…

## Main changes explained:
- Introduce new teamCode and findTeammatesForUser in src/controllers/teamController.js
- Update function putUserProfile in the src/controllers/userProfileController.js to include new logic for team update
- Implement new getUserSummaries function in src/controllers/userProfileController.js
- Added two functions: getAllSummariesByUserId and getTeammatesForUser in src/helpers/userHelper.js 
- Created index for user id in Team model in src/models/team.js
- Added new API route in src/routes/teamRouter.js and src/routes/userProfileRouter.js

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
-  Please refer to [2110 ](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/2110) frontend PR.

## Screenshots or videos of changes:


## Note:
Include the information the reviewers need to know.
